docs: Add guide for snapshot_socket. (#3295)
inspektor-gadget/inspektor-gadget: gadgets/snapshot_socket/README.mdx

---
title: snapshot_socket

> ---

Reviewer: Please add a sidebar_position as done in https://github.com/inspektor-gadget/inspektor-gadget/commit/2e64540e15a111e92feb89d6140a0546de85b573 .

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

The `snapshot_socket` shows existing sockets.

## Getting started

Running the gadget:

<Tabs groupId="env">
    <TabItem value="kubectl-gadget" label="kubectl gadget">
        ```bash
        $ kubectl gadget run ghcr.io/inspektor-gadget/gadget/snapshot_socket:latest [flags]
        ```
    </TabItem>

    <TabItem value="ig" label="ig">
        ```bash
        $ sudo ig run ghcr.io/inspektor-gadget/gadget/snapshot_socket:latest [flags]
        ```
    </TabItem>
</Tabs>

## Guide

> Run a pod / container:

Reviewer 2: ```suggestion
Run a nginx application:
```

<Tabs groupId="env">
<TabItem value="kubectl-gadget" label="kubectl gadget">
```bash
$ kubectl run --restart=Never --image=nginx test-snapshot-socket
pod/test-snapshot-socket created
```
</TabItem>

<TabItem value="ig" label="ig">
```bash
$ docker run --name test-snapshot-socket -d nginx
...
```
</TabItem>
</Tabs>

> Then, run the gadget and see how it shows the sleep socket:

Reviewer 2: ```suggestion
Then, run the gadget and see how it shows the listening socket:
```

> Then, run the gadget and see how it shows the sleep socket:

Reviewer: ```suggestion
Then, run the gadget and see how it shows the socket used by nginx:
```
